dispatcher/dispatcher.py

```diff
@@ -31,52 +25,48 @@ def main():
-    # Declare database object
-    MongoConnector = MongoConnect(config, logger)
+    config = config['DEFAULT' if not args.test else "TESTING"]
+    config['MasterDAQConfig'] = json.loads(config['MasterDAQConfig'])
+    control_mc = daqnt.get_client('daq')
+    runs_mc = daqnt.get_client('runs')
+    logger = daqnt.get_daq_logger(config['LogName'], level=args.log, mc=control_mc)
+    vme_config = json.loads(config['VMEConfig'])
+
+    # Declare necessary classes
+    sh = daqnt.SignalHandler()
+    Hypervisor = daqnt.Hypervisor(control_mc[config['ControlDatabaseName']], logger,
+            config['MasterDAQConfig']['tpc'], vme_config, sh=sh, testing=args.test)
+    MongoConnector = MongoConnect(config, logger, control_mc, runs_mc, Hypervisor, args.test)
+    DAQControl = DAQController(config, MongoConnector, logger, Hypervisor)
 
     sleep_period = int(config['PollFrequency'])
 
     logger.info('Dispatcher starting up')
 
     while(sh.event.is_set() == False):
+        sh.event.wait(sleep_period)
         # Get most recent goal state from database. Users will update this from the website.
         if (goal_state := MongoConnector.get_wanted_state()) is None:
             continue
         # Get the Super-Detector configuration
-        current_config = MongoConnector.GetSuperDetector(goal_state)
+        current_config = MongoConnector.get_super_detector(goal_state)
         # Get most recent check-in from all connected hosts
-        if MongoConnector.GetUpdate(current_config):
+        if (latest_status := MongoConnector.get_update(current_config)) is None:
             continue
-        latest_status = MongoConnector.latest_status
 
         # Print an update
         for detector in latest_status.keys():
             state = 'ACTIVE' if goal_state[detector]['active'] == 'true' else 'INACTIVE'
-            #linked_mode = 'ALL_LINKED' if goal_state['tpc']['link_mv'] == 'true' and goal_state['tpc']['link_nv'] == 'true' else 'NOLINK'
-            if goal_state['tpc']['link_mv'] == 'true' and goal_state['tpc']['link_nv'] == 'true':
-                linked_mode = 'ALL_LINKED'
-            elif goal_state['tpc']['link_mv'] == 'false' and goal_state['tpc']['link_nv'] == 'false':
-                linked_mode = 'NO_LINK'
-            elif goal_state['tpc']['link_mv'] == 'true' and goal_state['tpc']['link_nv'] == 'false':
-                linked_mode = 'MV_TPC_LINKED'
-            elif goal_state['tpc']['link_mv'] == 'false' and goal_state['tpc']['link_nv'] == 'true':
-                linked_mode = 'NV_TPC_LINKED'
-            else:
-                linked_mode = 'unclear'
-
             msg = (f'The {detector} should be {state} and is '
-                   f'{latest_status[detector]["status"].name}'
-                   f' and is in the linked mode {linked_mode}')
+                    f'{latest_status[detector]["status"].name}')
+            # TODO add statement about linking
```

Comment:
We could check on the type of detectors: if we only have the tpc we can print "all linked"; if we have tpc and neutron_veto we can print "tpc-mv linked, nv unlinked", and so on... ?